org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xinclude/XIncludeErrorCode.java

```diff
@@ -0,0 +1,135 @@
+	MultipleFallbacks("MultipleFallbacks"), //
+	FallbackParent("FallbackParent"), //
+	IncludeChild("IncludeChild"), //
+	FallbackChild("FallbackChild"), //
+	HrefMissing("HrefMissing"), //
+	RecursiveInclude("RecursiveInclude"), //
+	InvalidParseValue("InvalidParseValue"), //
+	XMLParseError("XMLParseError"), //
+	XMLResourceError("XMLResourceError"), //
+	TextResourceError("TextResourceError"), //
+	NonDuplicateNotation("NonDuplicateNotation"), //
+	NonDuplicateUnparsedEntity("NonDuplicateUnparsedEntity"), //
+	XpointerMissing("XpointerMissing"), //
+	AcceptMalformed("AcceptMalformed"), //
+	AcceptLanguageMalformed("AcceptLanguageMalformed"), //
+	RootElementRequired("RootElementRequired"), //
+	MultipleRootElements("MultipleRootElements"), //
+	ContentIllegalAtTopLevel("ContentIllegalAtTopLevel"), //
+	UnexpandedEntityReferenceIllegal("UnexpandedEntityReferenceIllegal"), //
+	HrefFragmentIdentifierIllegal("HrefFragmentIdentifierIllegal"), //
+	HrefSyntacticallyInvalid("HrefSyntacticallyInvalid"), //
+	XPointerStreamability("XPointerStreamability"), //
+	XPointerResolutionUnsuccessful("XPointerResolutionUnsuccessful");
+
+	private final String code;
+
+	private XIncludeErrorCode() {
+		this(null);
+	}
+
+	private XIncludeErrorCode(String code) {
+		this.code = code;
+	}
+
+	@Override
+	public String getCode() {
+		if (code == null) {
+			return name();
+		}
+		return code;
+	}
+
+	private final static Map<String, XIncludeErrorCode> codes;
+
+	static {
+		codes = new HashMap<>();
+		for (XIncludeErrorCode errorCode : values()) {
+			codes.put(errorCode.getCode(), errorCode);
+		}
+	}
+
+	public static XIncludeErrorCode get(String name) {
+		return codes.get(name);
+	}
+
+	/**
+	 * Create the LSP range from the SAX error.
+	 *
+	 * @param characterOffset
+	 * @param key
+	 * @param arguments
+	 * @param document
+	 * @return the LSP range from the SAX error.
+	 */
+	public static Range toLSPRange(XMLLocator location, XIncludeErrorCode code, Object[] arguments,
+			DOMDocument document) {
+		int offset = location.getCharacterOffset() - 1;
+		// adjust positions
+		switch (code) {
+			case NoFallback:
+			case XMLResourceError:
+				return XMLPositionUtility.selectAttributeValueAt("href", offset, document);
+			case MultipleFallbacks:
+			case FallbackParent:
+			case IncludeChild:
+			case FallbackChild:
+			case HrefMissing:
+			case RecursiveInclude:
+			case InvalidParseValue:
+			case XMLParseError:
+			case TextResourceError:
+			case NonDuplicateNotation:
+			case NonDuplicateUnparsedEntity:
+			case XpointerMissing:
+			case AcceptMalformed:
+			case AcceptLanguageMalformed:
+			case RootElementRequired:
+			case MultipleRootElements:
+			case ContentIllegalAtTopLevel:
+			case UnexpandedEntityReferenceIllegal:
+			case HrefFragmentIdentifierIllegal:
+			case HrefSyntacticallyInvalid:
+			case XPointerStreamability:
+			case XPointerResolutionUnsuccessful:
+				return XMLPositionUtility.selectStartTagName(offset, document);
```

Comment:
I wonder if all error code should higlight start tag name? If no, we can leave like this, but I wonder if we could have a test per XIInclude error code (or the most commons error code with xiinclude)? If you think it is too many long to do, let's give up my comments.